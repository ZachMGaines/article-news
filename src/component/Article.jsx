import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ author, title, description, url, urlToImage }) => (
  <figure>
    <img src={urlToImage} alt={title} />
    <figcaption>
      <p>{title}</p>
      <p>{author}</p>
      <p>{description}</p>
      <p>{urlToImage}</p>
      <p>{url}</p>
    </figcaption>
  </figure >
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  urlToImage: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Article;
