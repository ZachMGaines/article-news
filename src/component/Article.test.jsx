/**
 * @jest-environment jsdom
 */


import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  it('renders Article', () => {
    const { asFragment } = render(
      <Article
        title="Test title"
        author="Test author"
        description="Test description"
        urlToImage="http"
        url="hi"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
