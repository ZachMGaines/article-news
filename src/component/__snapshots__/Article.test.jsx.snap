// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Test title"
      src="http"
    />
    <figcaption>
      <p>
        Test title
      </p>
      <p>
        Test author
      </p>
      <p>
        Test description
      </p>
      <p>
        http
      </p>
      <p>
        hi
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
